--- handler_logging.py	2024-09-04 07:37:46
+++ handler_logging.py	2024-09-04 07:37:42
@@ -92,14 +92,14 @@
         listener_uid = pwd.getpwnam('listener').pw_uid
         adm_gid = grp.getgrnam('adm').gr_gid
         old_uid = os.geteuid()
-        try:
-            if old_uid != 0:
-                listener.setuid(0)
-            os.chown(self.baseFilename, listener_uid, adm_gid)
-            os.chmod(self.baseFilename, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP)
-        finally:
-            if old_uid != 0:
-                listener.unsetuid()
+        # try:
+        #     if old_uid != 0:
+        #         listener.setuid(0)
+        #     os.chown(self.baseFilename, listener_uid, adm_gid)
+        #     os.chmod(self.baseFilename, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP)
+        # finally:
+        #     if old_uid != 0:
+        #         listener.unsetuid()


 class ModuleHandler(logging.Handler):
@@ -210,19 +210,20 @@
         listener_uid = pwd.getpwnam('listener').pw_uid
         adm_grp = grp.getgrnam('adm').gr_gid
         if not os.path.isdir(log_dir):
-            old_uid = os.geteuid()
-            try:
-                if old_uid != 0:
-                    listener.setuid(0)
-                os.mkdir(log_dir)
-                os.chown(log_dir, listener_uid, adm_grp)
-                os.chmod(
-                    log_dir,
-                    stat.S_ISGID | stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IRGRP | stat.S_IXGRP,
-                )
-            finally:
-                if old_uid != 0:
-                    listener.unsetuid()
+            os.mkdir(log_dir)
+            # old_uid = os.geteuid()
+            # try:
+            #     if old_uid != 0:
+            #         listener.setuid(0)
+            #     os.mkdir(log_dir)
+            #     os.chown(log_dir, listener_uid, adm_grp)
+            #     os.chmod(
+            #         log_dir,
+            #         stat.S_ISGID | stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IRGRP | stat.S_IXGRP,
+            #     )
+            # finally:
+            #     if old_uid != 0:
+            #         listener.unsetuid()
         _logger_cache[name] = get_listener_logger(logger_name, file_path)
     return _logger_cache[name]

