From 8e6c9467d5e3145b464d1ffd03a4c3d2bccc1cbf Mon Sep 17 00:00:00 2001
From: Johannes Bornhold <johannes.bornhold.extern@univention.de>
Date: Thu, 2 May 2024 12:46:21 +0200
Subject: [PATCH] allow to specify separate notifier address

---
 .../univention-directory-listener/src/main.c  | 22 +++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

diff --git a/management/univention-directory-listener/src/main.c b/management/univention-directory-listener/src/main.c
index cf4c90ca02b..abfc6236496 100644
--- a/management/univention-directory-listener/src/main.c
+++ b/management/univention-directory-listener/src/main.c
@@ -191,6 +191,9 @@ static void usage(void) {
 	fprintf(stderr, "   -F   run in foreground (intended for process supervision)\n");
 	fprintf(stderr, "   -H   LDAP server URI\n");
 	fprintf(stderr, "   -h   LDAP server address\n");
+	fprintf(stderr, "   -n   Notifier server address (Optional, and only needed in case\n"
+					"        the notifier server address is not the same as the\n"
+					"        'LDAP server address' which can be specified with -h)\n");
 	fprintf(stderr, "   -p   LDAP server port\n");
 	fprintf(stderr, "   -b   LDAP base dn\n");
 	fprintf(stderr, "   -D   LDAP bind dn\n");
@@ -287,7 +290,7 @@ static void prepare_cache(const char *cache_dir) {
 /* Open LDAP and Notifier connection.
  * @return 0 on success, 1 on error.
  */
-static int do_connection(univention_ldap_parameters_t *lp) {
+static int do_connection(univention_ldap_parameters_t *lp, char *notifier_host) {
 	LDAPMessage *res;
 	int rc;
 	char **_attrs = NULL;
@@ -303,7 +306,9 @@ static int do_connection(univention_ldap_parameters_t *lp) {
 		univention_debug(UV_DEBUG_LISTENER, UV_DEBUG_WARN, "can not connect to LDAP server %s:%d", lp->uri ? lp->uri : lp->host ? lp->host : "NULL", lp->port);
 		goto fail;
 	}
-	if (NOTIFIER_CLIENT_NEW_RETRY(notifier_client_new(NULL, lp->host, 1)) != 0)
+
+	notifier_host = notifier_host ? notifier_host : lp->host;
+	if (NOTIFIER_CLIENT_NEW_RETRY(notifier_client_new(NULL, notifier_host, 1)) != 0)
 		goto fail;

 	/* check if we are connected to an OpenLDAP */
@@ -331,6 +336,7 @@ fail:
 int main(int argc, char *argv[]) {
 	univention_ldap_parameters_t *lp;
 	univention_ldap_parameters_t *lp_local;
+	char *notifier_host = NULL;
 	char *server_role;
 	int debugging = 0;
 	bool from_scratch = false;
@@ -382,7 +388,7 @@ int main(int argc, char *argv[]) {
 	for (;;) {
 		int c;

-		c = getopt(argc, argv, "d:FH:h:p:b:D:w:y:xZY:U:R:Km:Bc:giol:P");
+		c = getopt(argc, argv, "d:FH:h:n:p:b:D:w:y:xZY:U:R:Km:Bc:giol:P");
 		if (c < 0)
 			break;
 		switch (c) {
@@ -398,6 +404,9 @@ int main(int argc, char *argv[]) {
 		case 'h':
 			lp->host = strdup(optarg);
 			break;
+		case 'n':
+			notifier_host = strdup(optarg);
+			break;
 		case 'p':
 			lp->port = atoi(optarg);
 			break;
@@ -502,7 +511,11 @@ int main(int argc, char *argv[]) {
 		select_server(lp);
 	}

-	while (do_connection(lp) != 0) {
+	if (NULL == notifier_host) {
+		notifier_host = strdup(lp->host);
+	}
+
+	while (do_connection(lp, notifier_host) != 0) {
 			if (initialize_only) {
 				univention_debug(UV_DEBUG_LISTENER, UV_DEBUG_ERROR, "can not connect any server, exit");
 				exit(1);
@@ -599,6 +612,7 @@ int main(int argc, char *argv[]) {

 	univention_ldap_close(lp);
 	univention_ldap_close(lp_local);
+	free(notifier_host);

 	exit_handler(0);
 }
--
2.44.0

